### ISSUE TITLE
Formatter error on reference after match statement

### ISSUE THREAD

**Issue by johnthagen** [intellij-rust/intellij-rust#831]
The formatter formats the following code:

```rust
enum Day {
    Monday
}

fn fun(x: &mut i32, day: Day) {
    match day {
        _ => {}
    }

    *x += 1;
}


fn main() {
    let mut x = 5;
    fun(&mut x, Day::Monday);
}
```

To:

```rust
enum Day {
    Monday
}

fn fun(x: &mut i32, day: Day) {
    match day {
        _ => {}
    }

        * x += 1;
}


fn main() {
    let mut x = 5;
    fun(&mut x, Day::Monday);
}
```

Seems to get the `*x += 1;` part wrong.

Tested on plugin 0.1.0.1243.

**Comment by matklad** [comment]
Closed by 74b7f68546341434d12b7528503482c6f91a81a4: this actually was a parser issue.

**Comment by matklad** [comment]
Thanks for the report!
